src/SingleSignOnDemo/SSOExample.cs

```diff
@@ -0,0 +1,129 @@
+        /// La clé publique de la plate-forme (fourni par SMS digital).
+        /// </summary>
+        private const string PLATFORM_PUBLIC_KEY = "sQsT5C+Bst2i+hbBZIrBroHhDy7LopgFwUR9PFprans=";
+
+        /// <summary>
+        /// Der private Schlüssel der Anwendung (vom Anwendungsentwickler generiert).
+        /// The private key of the application (generated by the application developer).
+        /// La clé privée de l'application (généré par le promoteur de l'application).
+        /// </summary>
+        private const string APPLICATION_PRIVATE_KEY = "I2D6BWycz+bNWkcS1ul820uK5Jx1qGvsRfilDy09o94=";
+
+        public SSOExample()
+        {
+            // Keine Initialisierung notwendig.
+            // No initialization needed.
+            // Aucun besoin d'initialisation.
+        }
+
+        /// <summary>
+        /// Führt einen Login-Prozess mit dem übergebenen Token durch.
+        /// Executes the login process with the given token.
+        /// Executer le procès d'authentification avec le jeton donné.
+        /// </summary>
+        /// <param name="token">Das Token, das für den Login verwendet werden soll. The token used to execute the login process. Le signe utilisé pour executer le procès d'authentification.</param>
+        /// <returns></returns>
+        public static bool Login(string token)
+        {
+            try
+            {
+                // Eine Nonce erstellen, um ein Box-Objekt erstellen zu können.
+                // Create a nonce used to create a Box object.
+                // Créer un nonce utilisé pour créer un objet Box.
+                byte[] nonce = PublicKeyBox.GenerateNonce();
+
+                // Einen 64 Byte großen Puffer erstellen und ihn mit zufälligen Bytes füllen.
+                // Create a 64-byte buffer array and fill it with random bytes.
+                // Créer un tableau tampon de 64 octets et remplisser-le avec des octets aléatoires.
+                byte[] buffer = new byte[64];
+                new Random().NextBytes(buffer);
+
+                // Ein Box-Object erzeugen, die zur Kommunikation mit mySMS unter Verwendung des öffentlichen und privaten Schlüssels dient.
+                // Create a Box object to communicate with mySMS using the public and private key.
+                // Créer un object Box pour communiquer avec mySMS en utilisant les clés publique et privée.
+                byte[] encrypted = PublicKeyBox.Create(buffer, nonce, Convert.FromBase64String(APPLICATION_PRIVATE_KEY), Convert.FromBase64String(PLATFORM_PUBLIC_KEY));
+
+                // Nonce und verschlüsselten Text aneinanderreihen und in einen Base64-String umwandeln.
+                // Concatenate the nonce and the encrypted text and convert it to a base64 string.
+                // Concaténer le nonce et le texte chiffré et convertisser-le en chaîne base64.
+                byte[] payload = nonce.ToList().Concat(encrypted.ToList()).ToArray();
```

Comment:
creating a list from an array just to have the handy `concat` method feels a bit hacky to me. Did you consider using something like:
```
byte payload = new byte[nonce.Length + encrypted.Length];
nonce.CopyTo(payload, 0);
encrypted.CopyTo(payload, nonce.Length);